actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -32,5 +44,13 @@ def method_for_action(action_name)
         "default_render"
       end
     end
+
+    private
+      def no_templates?
+        # Check and see if any templates exist for the given action name, and prefixes.
+        # Variants are provided also as an argument, to check in case a template exists
+        # for a format not specifically declared in the `respond_to` block.
+        !template_exists_in_any_format?(action_name.to_s, _prefixes, { variants: request.variant })
```

Comment:
I'm still not sure why we need to pass the variant here; isn't the point of `template_exists_in_any_format?` to return true if *any* template (for any formats or variant or not) exists? Why are we taking into account the variant in the lookup? Is this a technical limitation of the architecture?

By the way, does this work with variants? Probably good to add some tests for that too